tidy3d/plugins/adjoint/components/base.py

```diff
@@ -37,6 +37,16 @@ def tree_flatten(self) -> Tuple[list, dict]:
             sub_children, sub_aux_data = jax_tree_flatten(field)
             children.append(sub_children)
             aux_data[field_name] = sub_aux_data
+
+        # fixes bug with jax handling 2D numpy array in polyslab vertices
+        if aux_data.get("type", "") == "JaxSimulation":
+            structures = aux_data["structures"]
+            for i, structure in enumerate(structures):
+                geometry = structure["geometry"]
+                if geometry["type"] == "PolySlab":
+                    vertices = geometry["vertices"]
+                    aux_data["structures"][i]["geometry"]["vertices"] = vertices.tolist()
```

Comment:
This probably still leaves the problem that `geometry` is a `GeometryGroup` that contains `PolySlab`-s. More generally there is also the new boolean operations that have not been tested with adjoint either. To handle this, Lucas has introduced this convenience method though: https://github.com/flexcompute/tidy3d/blob/2e07342b57a2a55efd2c54f37601cc317d8c211e/tidy3d/components/geometry/utils.py#L46-L67. I think something similar is what needs to be used here (but not sure if it can be directly used...)

Reply:
good point. I think I'll have to pick this up on Monday unfortunately.

Reply:
I just gave it a shot and it worked. just made the function recursive.

Reply:
So technically this solves geometry group but not clip operation...

Reply:
Oh right, so I just added a special handling for clip operation. I think that covers everything?